File: cormen/python/selection-sort.py
a = [11, 10, 23, 1000, 44, 3]
b = [5, 4, 3, 2 ,1]
def selectionSort(a):
    print('Before', a)
    for i in range(0, len(a) - 1):
        minIndex = i
        for j in range(i+1, len(a)):
            if(a[j] < a[minIndex]):
                minIndex = j
        if(minIndex != i):
            tempVar = a[i]
            a[i] = a[minIndex]
            a[minIndex] = tempVar
        print('During', a)
    print('After ', a)

selectionSort(a)

File: cormen/c/selection-sort.c
#include <stdio.h>
#include "arrayfunc.h"

void selectionSort(int *arr, int size) {
  int i, j, minIndex, tempVar;
  for(i = 0; i < size; i++){
    minIndex = i;
    for(j = i + 1; j < size; j++) {
      if(arr[minIndex] > arr[j]) {
        minIndex = j;
      }
    }
    if(minIndex != i){
      tempVar = arr[i];
      arr[i] = arr[minIndex];
      arr[minIndex] = tempVar;
    }
  }
}

void main() {
  int arr[] = {10, 11, 50, 3, 2, 40, 1, -1};
  int size = sizeof(arr)/sizeof(arr[0]);
  printArray(arr, size);
  selectionSort(arr, size);
  printArray(arr, size);
}
